rmw_fastrtps_shared_cpp/src/publisher.cpp

```diff
@@ -38,17 +33,41 @@ destroy_publisher(
   assert(publisher->implementation_identifier == identifier);
   static_cast<void>(identifier);
 
+  // Get RMW Publisher
+  rmw_ret_t ret = RMW_RET_OK;
   auto info = static_cast<CustomPublisherInfo *>(publisher->data);
-  Domain::removePublisher(info->publisher_);
-  delete info->listener_;
 
-  _unregister_type(participant_info->participant, info->type_support_);
-  delete info;
+  if (nullptr != info) {
```

Comment:
@MiguelCompany as mentioned elsewhere, do not partially finalize, do not put guards against bogus calling code.